src/vm/i386/stublinkerx86.cpp

```diff
@@ -6606,12 +6626,13 @@ void FixupPrecode::ResetTargetInterlocked()
 
     PCODE target = (PCODE)GetEEFuncEntryPoint(PrecodeFixupThunk);
     MethodDesc* pMD = (MethodDesc*)GetMethodDesc();
-    newValue.m_rel32 =
 #ifdef FIXUP_PRECODE_PREALLOCATE_DYNAMIC_METHOD_JUMP_STUBS
-        pMD->IsLCGMethod() ?
-            rel32UsingPreallocatedJumpStub(&m_rel32, target, GetDynamicMethodEntryJumpStub()) :
+    // The entry point of LCG methods cannot revert back to the original entry point, as their jump stubs would have to be
+    // reused, which is currently not supported
+    _ASSERTE(!pMD->IsLCGMethod());
```

Comment:
Just out of curiosity, is there anything that prevents the tiered compilation stuff from kicking in for LCG methods, or is this an assert that might be reachable in tired compilation cases?

Reply:
The eligibility for tiered compilation is controlled by: `MethodDesc::IsEligibleForTieredCompilation()`

Reply:
cc @noahfalk

Reply:
Yes this path is currently only reached from tiered compilation and it is disabled for LCG methods in the function above